### PR TITLE
KTOR-3031 Fix autoreload not working by default under Windows

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-3031](https://youtrack.jetbrains.com/issue/KTOR-3031) Autoreload not working by default under Windows

### CODE REVIEW

**Review by osipxd — APPROVED** [review]
LGTM! Just a couple of minor comments

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/engine/EmbeddedServerJvm.kt [hunk 1]
```diff
@@ -388,3 +388,9 @@ actual constructor(
         }
     }
 }
+
+internal fun checkUrlMatches(url: URL, pattern: String): Boolean {
+    val urlPath = url.path?.replace(File.separatorChar, '/')
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-server/ktor-server-core/jvm/src/io/ktor/server/engine/EmbeddedServerJvm.kt]: if explicit set

File: ktor-server/ktor-server-core/jvm/test/io/ktor/server/engine/EmbeddedServerTest.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+package io.ktor.server.engine
+
+import org.junit.jupiter.api.*
+import org.junit.jupiter.api.Assertions.*
+import java.net.*
+
+class EmbeddedServerTest {
+
+    @Test
+    fun `checkUrlMatches returns true when pattern matches part of URL path`() {
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-server/ktor-server-core/jvm/test/io/ktor/server/engine/EmbeddedServerTest.kt:14]: I would keep it with separate cases cause it's easier to re-run in IDE (but can change if you insist :) )

[the remark this answers is not in the document]
Comment (osipxd) [reply ktor-server/ktor-server-core/jvm/test/io/ktor/server/engine/EmbeddedServerTest.kt:14]: I don't insist :)

File: ktor-server/ktor-server-core/jvm/src/io/ktor/server/engine/EmbeddedServerJvm.kt [hunk 3]
```diff
@@ -388,3 +388,9 @@ actual constructor(
         }
     }
 }
+
+internal fun checkUrlMatches(url: URL, pattern: String): Boolean {
+    val urlPath = url.path?.replace(File.separatorChar, '/') ?: return false
+    val normalizedPattern = pattern.replace(File.separatorChar, '/')
+    return urlPath?.contains(normalizedPattern, ignoreCase = true) == true
```
Comment (osipxd) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/server/engine/EmbeddedServerJvm.kt:395]: One more minor change
```suggestion
    return urlPath.contains(normalizedPattern, ignoreCase = true)
```

**Review by osipxd — APPROVED** [review]

**Pull request merged**